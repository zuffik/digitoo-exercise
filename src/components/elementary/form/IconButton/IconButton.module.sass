@import "../../../../styles/variables"

.btn
  background-color: transparent
  border-radius: 0
  outline: none !important
  border: none !important
  padding: $spacer

