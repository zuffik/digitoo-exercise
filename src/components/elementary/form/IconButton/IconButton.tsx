import * as React from 'react';
import classNames from 'classnames';
import styles from './IconButton.module.sass';

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
}

export const IconButton: React.FC<Props> = (props: Props) => {
    return (
        <button {...props} className={classNames(props.className, styles.btn)}/>
    );
};
