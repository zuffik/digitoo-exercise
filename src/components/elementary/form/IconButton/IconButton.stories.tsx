import * as React from 'react';
import {FaPen} from 'react-icons/fa';
import {IconButton} from './IconButton';
import {action} from '@storybook/addon-actions';

export default {
  title: 'Elementary/Form',
};

export const iconButton = () => (
  <IconButton onClick={action('onClick')}>
    <FaPen />
  </IconButton>
);
