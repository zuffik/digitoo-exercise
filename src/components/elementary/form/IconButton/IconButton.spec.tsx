import * as React from 'react';
import {render} from '@testing-library/react'
import {IconButton} from "./IconButton";

describe('<IconButton/>', () => {
    it('should render', () => {
        const {asFragment} = render(<IconButton/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
