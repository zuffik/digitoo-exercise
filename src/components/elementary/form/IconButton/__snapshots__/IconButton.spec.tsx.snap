// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton/> should render 1`] = `
<DocumentFragment>
  <button
    class="btn"
  />
</DocumentFragment>
`;
